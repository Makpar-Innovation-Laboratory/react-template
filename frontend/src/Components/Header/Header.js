import React from 'react';
// import { Context } from "../../App";
import { Link, useLocation } from "react-router-dom";

export default function Navigation() {
    const location = useLocation();

    return (
        <div className="d-flex flex-column">
            {/* Main header */}
            <div className="nav-container background-light d-flex justify-content-between align-items-center py-4">
                <div className="nav-header px-5">
                    <h1 className="color-dark">LOGO</h1>
                </div>

                <div className="px-5">
                    <a className=" color-dark">
                        <h5 className="color-dark logout">Sign out</h5>
                    </a>
                </div>
            </div>

            {/* Navigation */}
            <div className="navigation-items d-flex justify-content-center align-items-center background-dark">
                <a>
                    <div
                        className="navigation-item-white color-white"
                        id={location.pathname == "/" ? "active-white" : ""}
                    >
                        <Link to="/" className="color-white">
                            <h5 className="color-white">Home</h5>
                        </Link>
                    </div>
                </a>
                <a>
                    <div
                        className="navigation-item-white color-white"
                        // id={location.pathname == "/laborCategory" ? "active-white" : ""}
                    >
                        <Link to="/ToolOne" className="color-white">
                            <h5 className="color-white">Tool 1 (Pie Chart)</h5>
                        </Link>
                    </div>
                </a>
                <a>
                    <div
                        className="navigation-item-white color-white"
                        // id={location.pathname == "/scheduleManagement" ? "active-white" : ""}
                    >
                        <Link to="/ToolTwo" className="color-white">
                            <h5 className="color-white">Tool 2 (Area Chart)</h5>
                        </Link>
                    </div>
                </a>
                <a>
                    <div
                        className="navigation-item-white color-white"
                        // id={location.pathname == "/scheduleManagement" ? "active-white" : ""}
                    >
                        <Link to="/ToolThree" className="color-white">
                            <h5 className="color-white">Tool 3 (2 step)</h5>
                        </Link>
                    </div>
                </a>
                <a>
                    <div
                        className="navigation-item-white color-white"
                        // id={location.pathname == "/scheduleManagement" ? "active-white" : ""}
                    >
                        <Link to="/ToolFour" className="color-white">
                            <h5 className="color-white">Tool 4 (Single page)</h5>
                        </Link>
                    </div>
                </a>
            </div>
        </div>
    )
}