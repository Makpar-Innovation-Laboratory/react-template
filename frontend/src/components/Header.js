import React, { useState }  from "react";
import Navigation from "./Navigation";
import { Link } from "react-router-dom";

const Header = () => {
  const [isExpanded, expandBanner] = useState(false);

  return (
    <div className="nav-bar">
      <h1>MAKPAR</h1>
      <nav className="nav-items">
        <Link to="/" className="nav-link">HOME</Link>
        <Link to="/mission" className="nav-link">MISSION</Link>
        <Link to="/team" className="nav-link">TEAM</Link>
        <Link to="/projects" className="nav-link">PROJECTS</Link>
        <Link to="/post-archive" className="nav-link">ARCHIVE</Link>
        {/* <a routerLink="/home" class="nav-link">HOME</a>
        <a routerLink="/mission" class="nav-link">MISSION</a>
        <a routerLink="/" class="nav-link">TEAM</a>
        <a routerLink="/" class="nav-link">PROJECTS</a>
        <a routerLink="/" class="nav-link">POST ARCHIVE</a> */}
      </nav>
    </div>
  );

};
export default Header;
