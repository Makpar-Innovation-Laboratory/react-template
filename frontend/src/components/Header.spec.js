import React from "react";
import { isolateComponent } from "isolate-react";
import Header from "./Header";

describe("JSX Tests :: <Header /> Rendering", () => {

  test("Sanity Check Snapshot Rendering", () => {
    const wrapper = isolateComponent(<Header />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  test("Explanation Banner Shows When Toggled", () => {
    const wrapper = isolateComponent(<Header />);
    const toggleBtn = wrapper.findOne("#explanation-btn");
    const explanation = wrapper.findAll("#gov-banner");

    expect(explanation.length).toEqual(0);

    toggleBtn.props.onClick();
    const explanationAfter = wrapper.findAll("#gov-banner");

    expect(explanationAfter.length).toEqual(1);
  });
});
