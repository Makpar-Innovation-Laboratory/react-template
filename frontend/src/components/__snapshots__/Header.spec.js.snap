// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX Tests :: <Header /> Rendering Sanity Check Snapshot Rendering 1`] = `
"{\\"nodeType\\":\\"html\\",\\"type\\":\\"div\\",\\"name\\":\\"div\\",\\"props\\":{\\"className\\":\\"nav-bar\\",\\"children\\":[{\\"type\\":\\"h1\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"MAKPAR\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"nav\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"className\\":\\"nav-items\\",\\"children\\":[{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"HOME\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/mission\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"MISSION\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/team\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"TEAM\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/projects\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"PROJECTS\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/archive\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"ARCHIVE\\"},\\"_owner\\":null,\\"_store\\":{}}]},\\"_owner\\":null,\\"_store\\":{}}]},\\"key\\":\\"null\\"}
 -> {\\"nodeType\\":\\"html\\",\\"type\\":\\"h1\\",\\"name\\":\\"h1\\",\\"props\\":{\\"children\\":\\"MAKPAR\\"},\\"key\\":\\"null\\"}
  -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"MAKPAR\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
 -> {\\"nodeType\\":\\"html\\",\\"type\\":\\"nav\\",\\"name\\":\\"nav\\",\\"props\\":{\\"className\\":\\"nav-items\\",\\"children\\":[{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"HOME\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/mission\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"MISSION\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/team\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"TEAM\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/projects\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"PROJECTS\\"},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"to\\":\\"/archive\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"ARCHIVE\\"},\\"_owner\\":null,\\"_store\\":{}}]},\\"key\\":\\"null\\"}
  -> {\\"nodeType\\":\\"react\\",\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"name\\":\\"Link\\",\\"props\\":{\\"to\\":\\"/\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"HOME\\"},\\"key\\":\\"null\\"}
   -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"HOME\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
  -> {\\"nodeType\\":\\"react\\",\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"name\\":\\"Link\\",\\"props\\":{\\"to\\":\\"/mission\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"MISSION\\"},\\"key\\":\\"null\\"}
   -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"MISSION\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
  -> {\\"nodeType\\":\\"react\\",\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"name\\":\\"Link\\",\\"props\\":{\\"to\\":\\"/team\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"TEAM\\"},\\"key\\":\\"null\\"}
   -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"TEAM\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
  -> {\\"nodeType\\":\\"react\\",\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"name\\":\\"Link\\",\\"props\\":{\\"to\\":\\"/projects\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"PROJECTS\\"},\\"key\\":\\"null\\"}
   -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"PROJECTS\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
  -> {\\"nodeType\\":\\"react\\",\\"type\\":{\\"displayName\\":\\"Link\\",\\"propTypes\\":{}},\\"name\\":\\"Link\\",\\"props\\":{\\"to\\":\\"/archive\\",\\"className\\":\\"nav-link\\",\\"children\\":\\"ARCHIVE\\"},\\"key\\":\\"null\\"}
   -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"ARCHIVE\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}"
`;
